draft-documents/encrypted-data-vaults.md

```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
-application, whether a valid or invalid data format, encrypted or not, so long
-as a valid bearer token is included in the request. Clients interact with Gaia
-through a REST API.
-
-Identity Hubs enable the end user to store their data in one or more locations
-which are linked to a DID. Creating and updating data and metadata in the Hub is
-done by posting JWTs to specified endpoints. The JWT headers include metadata
-for the data in the JWT body, as well as a signature. The read and write
-requests are encrypted on the wire, and data is encrypted on the Hub. Access
-control is carried out via posting to the Permissions interface and indexing is
-done via the Collections interface. Clients are responsible for writing
-appropriate metadata to Collections, which are not themselves encrypted,
-enabling the Hub to respond to queries. Reads require multiple requests, first
-to retrieve metadata for the desired data object(s), and then to retrieve the
-sequence of commits which make up the actual data. Mechanisms for authentication
-and synchronization of changes and conflict resolution between Hub instances are
-still under development. Identity Hubs are responsible for additional things
-beyond just data storage, for example management of the end user's profile;
-transmission of human- or machine-readable messages through the Actions
-interface; pointers to external services.
-
-At a lower level, [Tahoe-LAFS](https://tahoe-lafs.org/trac/tahoe-lafs) uses a
-client-server architecture, whereby a local client encrypts data, breaks it into
-pieces, and stores it on one or more servers on other devices. The client
-creates and stores redundant pieces of data so a system is resilient to
-corruption of some shards or servers going offline. Servers can neither read nor
-modify the data. Data is organized in a filesystem like directory structure and
-access control makes use of this. Data can be mutable or immutable, and is
-always encrypted.
-
-Datashards is concerned with encrypted storage and replication of data across
-devices. Datashards is similar to Tahoe-LAFS, but uses a more generalized URI
-scheme and object capabilities for sharing of data objects. Data is always
-encrypted.
+users to sign into client applications, which can then read or write data.
+Resources (data objects) on Solid servers are represented by HTTP URIs, and Solid
+servers receive HTTP requests containing RDF payloads and create or modify the
+target URI accordingly.
+
+Data Hubs uses JSON Web Tokens (JWTs) and specified endpoints. Multiple requests
+are required, first to retrieve metadata for the desired data object(s), and then
+to retrieve the sequence of commits which make up the actual data. Mechanisms for
+authentication are still under development. Access control is carried out via
+posting to the Permissions interface.
+
+Tahoe-LAFS uses a client-server architecture, whereby a local client encrypts
```

Comment:
How do you feel about:

Tahoe-LAFS uses a client-gateway-storage server architecture, whereby a client passes data to a gateway server for encryption and chunking. The gateway, in turn, stores the individual chunks on a cluster of storage servers (several copies of the data are stored, for greater availability and disaster recovery). Services are identified ... [etc]